pyscriptjs/tests/integration/test_zz_examples.py

```diff
@@ -105,32 +105,51 @@ def test_bokeh_interactive(self):
         assert self.page.title() == "Bokeh Example"
         wait_for_render(self.page, "*", '<div.*?class=\\"bk\\".*?>')
 
-    @pytest.mark.xfail(reason="Flaky test #759")
     def test_d3(self):
-        # XXX improve this test
         self.goto("examples/d3.html")
         self.wait_for_pyscript()
         assert (
             self.page.title() == "d3: JavaScript & PyScript visualizations side-by-side"
         )
         wait_for_render(self.page, "*", "<svg.*?>")
+        assert "PyScript version" in self.page.content()
+        pyscript_chart = self.page.wait_for_selector("#py")
+
+        # Let's simply assert that the text of the chart is as expected which
+        # means that the chart rendered successfully and with the right text
+        assert "🍊21\n🍇13\n🍏8\n🍌5\n🍐3\n🍋2\n🍎1\n🍉1" in pyscript_chart.inner_text()
 
     def test_folium(self):
-        # XXX improve this test
         self.goto("examples/folium.html")
         self.wait_for_pyscript()
         assert self.page.title() == "Folium"
         wait_for_render(self.page, "*", "<iframe srcdoc=")
+        # assert "Unemployment Rate (%)" in self.page.content()
+
+        # We need to look into the iframe first
+        iframe = self.page.frame_locator("iframe")
+
+        # Just checking that legend was rendered correctly
+        legend = iframe.locator("#legend")
+        assert "Unemployment Rate (%)" in legend.inner_html()
+
+        # Let's check that the zoom buttons are rendered and clickable
+        # Note: if element is not clickable it will timeout
+        zoom_in = iframe.locator("[aria-label='Zoom in']")
+        assert "+" in zoom_in.inner_text()
+        zoom_in.click()
+        zoom_out = iframe.locator("[aria-label='Zoom out']")
+        assert "−" in zoom_out.inner_text()
+        zoom_out.click()
 
     def test_matplotlib(self):
-        # XXX improve this test
+        # Can't do much here since we are rendering an image
```

Comment:
One approach could be to compare the underlying numpy data for the image rendered through the canvas along with a reference image uploaded to the repository.

Reply:
This is the approach used in https://blog.pyodide.org/posts/canvas-renderer-matplotlib-in-pyodide/

A reference could be: https://github.com/pyodide/pyodide/blob/b0dc19e4a169c9dcf97e21fe0ac2c635a0900f13/packages/matplotlib/test_matplotlib.py#L72

Reply:
Oh awesome! Thank you for pointing me in the right direction, this is really cool :smile:

Reply:
You can get the canvas pixel data like this: https://github.com/pyodide/matplotlib-pyodide/blob/fbfe472e6acaf081855d02c3a4e52d6cb6b7b03d/matplotlib_pyodide/html5_canvas_backend.py#L72

Reply:
Resolving since this is incorporated!